Cybersecurity_Tools/DEEPFAKE/extension/backend/main.py

```
@@ -0,0 +1,305 @@
+        "video_model": video_model is not None
+    }
+
+def decode_base64_media(data: str) -> bytes:
+    """Decode base64 media data with validation"""
+    if "," in data:
+        data = data.split(",")[1]
+    
+    try:
+        return base64.b64decode(data)
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=f"Invalid base64 data: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)